Platforms/QemuQ35Pkg/Plugins/QemuRunner/QemuRunner.py

```diff
@@ -90,8 +90,16 @@ def Runner(env):
         if path_to_os is not None:
             # Potentially dealing with big daddy, give it more juice...
             args += " -m 8192"
-            #args += " -hda \"" + path_to_os + "\""
-            args += " -drive format=raw,index=0,media=disk,file=\"" + path_to_os + "\""
+
+            storage_rule = {

```

Comment:
For new code additions, I suggest using f-strings so the substitution is less verbose and `pathlib` to get the extension via `.suffix` which is more straightforward to read.

Something like:

```python
file_extension = Path(path_to_os).suffix.lower().replace('"', '')

storage_rule = {
    ".vhd": f" -drive format=raw,index=0,media=disk,file=\"{path_to_os}\"",
    ".qcow2": f" -hda \"{path_to_os}\""
}.get(file_extension, None)
```

Makes it easier to read.